src/Agent.Plugins/GitSourceProvider.cs

```diff
@@ -530,8 +530,9 @@ public abstract class GitSourceProvider : ISourceProvider
             }
             else
             {
+                char dirSeparator = Path.DirectorySeparatorChar;
                 // delete the index.lock file left by previous canceled build or any operation cause git.exe crash last time.
-                string lockFile = Path.Combine(targetPath, ".git\\index.lock");
+                string lockFile = Path.Combine(targetPath, $".git{dirSeparator}index.lock");
```

Comment:
I have no idea why it was implemented this way, but you actually should not need the dirSeparator at all. Path.Combine() should do this for us.

It should have been:
```
string lockFile = Path.Combine(targetPath, ".git", "index.lock");
```

Reply:
Much cleaner. Looking at the `Path.Combine` documentation this should indeed work. I will make a new commit to reflect that.